docs/arch/adr-3-store-status.md

```diff
@@ -0,0 +1,122 @@
+|Publish                      |Published          |N/A                 |
+|Schedule                     |Unpublished changes|Scheduled           |
+|Unschedule                   |Unpublished changes|2i Approved         |
+|Discard Draft                |Published          |N/A                 |
+|Unpublish                    |Draft              |N/A                 |
+
+
+### Status assumptions
+
+- Can only publish if workflow state is 2i Approved
+- Can only schedule if workflow state is 2i Approved
+- Can only Claim Review if workflow state is Submitted for 2i and the Claimer is not the same user as the one that submitted for 2i
+- After 2i has been approved, the user can submit for 2i again
+- Can only delete a step-by-step if it’s never been published
+
+
+## Decision
+
+Create a new field in the database to store a value of `status` that allows values of:
+
+- draft
+- published
+- scheduled
+- submitted_for_2i
+- in_review
+- 2i_approved
+- 2i_changes_requested
+
+
+"draft" and "unpublished changes" mean the same thing to the user, so if they are consolidated it won't be necessary to keep track of separate publication and workflow states. The presence of a `published_at` date can be used to determine if something has been previously published, as it is now.
+
+That means the 2i workflow can be simplified.
+
+### 2i workflow
+
+|Action                                  |Starting Status |End Status          |
+|----------------------------------------|:--------------:|-------------------:|
+|Create new/update published step-by-step|N/A or published|draft               |
+|Makes changes to step-by-step           |draft           |draft               |
+|Submit for 2i                           |draft           |submitted_for_2i    |
+|Claim Review                            |submitted_for_2i|in_review           |
+|2i Approved                             |in_review       |2i_approved         |
+|2i Requests changes                     |in_review       |2i_changes_requested|
+|Publish                                 |2i_approved     |published           |
+|Schedule                                |2i_approved     |scheduled           |
+|Unschedule                              |scheduled       |2i_approved         |
+
+
+### Other actions
+
+- Delete: Only if step-by-step is not scheduled or been previously published
+- Discard Draft: Only if step-by-step is not scheduled and has been previously published
+- Unpublish: Only if step-by-step is not scheduled and has been previously published
+
+
+### Pros
+
+- It will be easy to expand the allowed statuses to include factcheck statuses.
+- It will be easier to query and filter step-by-steps by status.
+- The status can be used by internal change notes to give a better description of what has changed.
+
+### Cons
+
+- Extra validation will be needed to ensure that a status can be set. E.g Cannot set to `in_review` if there is no `reviewer` recorded.
```

Comment:
Can you add a section about what happens to existing step by steps?

Are we going to create a rake task to apply a `status` value for them?

Reply:
I've added a note to the consequences about needing to update existing step-by-steps. I haven't specifically mentioned a rake task because that will come out in the implementation.